homeassistant/components/group/media_player.py

```diff
@@ -0,0 +1,405 @@
+    DOMAIN,
+    PLATFORM_SCHEMA,
+    SERVICE_CLEAR_PLAYLIST,
+    SERVICE_MEDIA_NEXT_TRACK,
+    SERVICE_MEDIA_PAUSE,
+    SERVICE_MEDIA_PLAY,
+    SERVICE_MEDIA_PREVIOUS_TRACK,
+    SERVICE_MEDIA_SEEK,
+    SERVICE_MEDIA_STOP,
+    SERVICE_PLAY_MEDIA,
+    SERVICE_SHUFFLE_SET,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    SERVICE_VOLUME_MUTE,
+    SERVICE_VOLUME_SET,
+    SUPPORT_CLEAR_PLAYLIST,
+    SUPPORT_NEXT_TRACK,
+    SUPPORT_PAUSE,
+    SUPPORT_PLAY,
+    SUPPORT_PLAY_MEDIA,
+    SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_SEEK,
+    SUPPORT_SHUFFLE_SET,
+    SUPPORT_STOP,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP,
+    MediaPlayerEntity,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_SUPPORTED_FEATURES,
+    CONF_ENTITIES,
+    CONF_NAME,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import HomeAssistant, State
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType, EventType
+
+KEY_CLEAR_PLAYLIST = "clear_playlist"
+KEY_ON_OFF = "on_off"
+KEY_PAUSE_PLAY_STOP = "play"
+KEY_PLAY_MEDIA = "play_media"
+KEY_SHUFFLE = "shuffle"
+KEY_SEEK = "seek"
+KEY_TRACKS = "tracks"
+KEY_VOLUME = "volume"
+
+DEFAULT_NAME = "Media Group"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Required(CONF_ENTITIES): cv.entities_domain(DOMAIN),
+    }
+)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: Callable,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Media Group platform."""
+    async_add_entities([MediaGroup(config[CONF_NAME], config[CONF_ENTITIES])])
+
+
+class MediaGroup(MediaPlayerEntity):
+    """Representation of a Media Group."""
+
+    def __init__(self, name: str, entities: list[str]) -> None:
+        """Initialize a Media Group entity."""
+        self._name = name
+        self._state: str | None = None
+        self._supported_features: int = 0
+
+        self._entities = entities
+        self._features: dict[str, set[str]] = {
+            KEY_CLEAR_PLAYLIST: set(),
+            KEY_ON_OFF: set(),
+            KEY_PAUSE_PLAY_STOP: set(),
+            KEY_PLAY_MEDIA: set(),
+            KEY_SHUFFLE: set(),
+            KEY_SEEK: set(),
+            KEY_TRACKS: set(),
+            KEY_VOLUME: set(),
+        }
+
+    async def _on_state_change(self, event: EventType) -> None:
+        self.async_set_context(event.context)
+        await self.async_update_supported_features(
+            event.data.get("entity_id"), event.data.get("new_state")  # type: ignore
+        )
+        await self.async_update_state()
+
+    async def async_update_supported_features(
```

Comment:
This doesn't need to be a coroutine function since we don't await inside. Make it a callback and decorate it with `@callback`.

Reply:
Done